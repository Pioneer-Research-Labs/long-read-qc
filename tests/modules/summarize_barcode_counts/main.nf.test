nextflow_process {

    name "Test Process summarize_barcode_counts"
    script "modules/summarize_barcode_counts/main.nf"
    process "summarize_barcode_counts"

    test("Should summarize barcode counts") {

        when {
            params {
                outdir = outputDir
            }
            process {
                """
                input[0] = file('${projectDir}/test_data/sample_map.tsv')
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$outputDir/concatenated_barcode_counts.csv").match()
        }

    }

}
