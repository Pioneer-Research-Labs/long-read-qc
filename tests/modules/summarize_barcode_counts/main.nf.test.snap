{
    "Should summarize barcode counts": {
        "content": [
            "/Users/sandie/projects/pioneerlabs/long-read-qc/.nf-test/tests/b9947510f683093dd2a001a8d767eb3c/results/concatenated_barcode_counts.csv"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-02T14:51:02.379524"
    }
}